*   Multimídia - Vídeo (Exercício 1)

*   Felipe Daniel Dias dos Santos - 11711ECP004

*   Graduação em Engenharia de Computação - Faculdade de Engenharia Elétrica - Universidade Federal de Uberlândia

In [ ]:
#Vídeo I

#Declaração das bibliotecas necessárias, definição do diretório raíz e montagem do diretório
from google.colab import drive
import numpy as np
import cv2
import os

drive.mount('/content/gdrive')

root_path = '/content/gdrive/My Drive/Colab Notebooks/dados/'

#Carregamento do vídeo I
vid_1 = cv2.VideoCapture(root_path + 'brandenburg.avi')

#Obtendo a taxa de quadros do vídeo I a partir da flag "CAP_PROP_FRAME_COUNT"
n_frames = int(vid_1.get(cv2.CAP_PROP_FRAME_COUNT))
print("Quantidade de quadros:", n_frames)

#Obtendo o codec do vídeo I a partir da flag "CAP_PROP_FOURCC", que diz respeito ao tipo de compressão
#O retorno da função é o codec do arquivo, ou seja, um código que identifica a compressão do arquivo de vídeo
#Caso o arquivo não tenha compressão, o retorno será 0
cod = int(vid_1.get(cv2.CAP_PROP_FOURCC))
print("Codec:", cod)

#Obtendo o tamanho do vídeo I
vid_1_length = os.stat(root_path + 'brandenburg.avi').st_size
print("Tamanho do arquivo (em bytes):", vid_1_length)

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).
Quantidade de quadros: 5577
Codec: 877677894
Tamanho do arquivo (em bytes): 91907758


In [ ]:
#Podemos obter as mesmas informações do vídeo I com o comando "ffprobe" da biblioteca "ffmpeg"
!ffprobe 'gdrive/My Drive/Colab Notebooks/dados/brandenburg.avi'

ffprobe version 3.4.8-0ubuntu0.2 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
  configuration: --prefix=/usr --extra-version=0ubuntu0.2 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --enable-gpl --disable-stripping --enable-avresample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librubberband --enable-librsvg --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-li

In [ ]:
#Vídeo II

#Carregamento do vídeo II
vid_2 = cv2.VideoCapture(root_path + 'grb.avi')

#Obtendo a quantidade de quadros do vídeo II
n_frames = int(vid_2.get(cv2.CAP_PROP_FRAME_COUNT))
print("Quantidade de quadros:", n_frames)

#Obtendo o codec do vídeo II
cod = int(vid_2.get(cv2.CAP_PROP_FOURCC))
print("Codec:", cod)

#Obtendo o tamanho do vídeo II
vid_2_length = os.stat(root_path + 'grb.avi').st_size
print("Tamanho do arquivo (em bytes):", vid_2_length)

Quantidade de quadros: 836
Codec: 877677894
Tamanho do arquivo (em bytes): 1028984


In [ ]:
#Podemos obter as mesmas informações do o vídeo II com o comando "ffprobe" da biblioteca "ffmpeg"
!ffprobe 'gdrive/My Drive/Colab Notebooks/dados/grb.avi'

ffprobe version 3.4.8-0ubuntu0.2 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
  configuration: --prefix=/usr --extra-version=0ubuntu0.2 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --enable-gpl --disable-stripping --enable-avresample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librubberband --enable-librsvg --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-li

Como foi observado, as informações diferem em relação a quantidade de quadros: 5577 para o primeiro vídeo e 836 para o segundo. Porém, as informações referentes ao codec são idênticas, significando que esses dois arquivos de vídeo possuem o mesmo tipo de compressão. O codec desses arquivos (877677894) se refere ao tipo de compressão MPEG4, de acordo com a a lista de codecs. Além disso, o comando "ffprobe" confirma essa informação: o retorno dessa função nos diz que há compressão em ambos os vídeos, do tipo MPEG4. O tamanho dos arquivos também diferem: 91.9 MB para o primeiro vídeo e 1 MB para o segundo. O tamanho de um arquivo de vídeo é dado pelo produto da quantidade de quadros, a resolução de um quadro e a quantidade de bytes de um pixel. Para um vídeo em tons de cinza (um canal), cada pixel tem 1 byte. Para um vídeo com sistema de três canais, cada pixel tem 3 bytes. Mas, utilizando um sistema de compressão, os vídeos tem um tamanho menor a esse.